src/decision/src/LidarDecision.cpp

```diff
@@ -31,6 +32,143 @@ void LidarDecision::scanCallBack(const sensor_msgs::LaserScan::ConstPtr& raw_sca
+    angle_increment_steps = static_cast<int>((raw_scan->angle_max - raw_scan->angle_min) / raw_scan->angle_increment);
+
+    //ranges start at 0?
+    while ((raw_scan->ranges[left_side_value]>distance_used)||(left_side_value < angle_increment_steps/2)){
+        left_side_value++;
+    }
+    std::cout << "left_side " << left_side_value << std::endl;
+    while((raw_scan->ranges[angle_increment_steps - right_side_value] > distance_used)||(right_side_value < angle_increment_steps/2)){
+        right_side_value++;
+    }
+    std::cout << "right_side " << right_side_value << std::endl;
+    if (left_side_value > right_side_value) return 1;
+    else return (-1);
+
+}
+
+
+geometry_msgs::Twist LidarDecision::manage_twist_2(const sensor_msgs::LaserScan::ConstPtr& raw_scan) {
+
+    geometry_msgs::Twist vel_msg;
+    float dis_near= 5;
+    float dis_mid = 10;
+    float dis_far = 20;
+    int angle_increment_steps;
+    angle_increment_steps = static_cast<int>((raw_scan->angle_max - raw_scan->angle_min) / raw_scan->angle_increment);
+    int angle_value_exact_front;
+    angle_value_exact_front = angle_increment_steps / 2;
+    int angle_side_near = 1; //the area that would not be obstacle to motion
+    int angle_side_mid = 1;
+    int angle_side_far = 1;
+
+    //step 1 obstacle in near, mid or far range (inside impact region);
+    bool in_near = obstacle_in_range(raw_scan->angle_min + angle_side_near, raw_scan->angle_max - angle_side_near,
+                                     dis_near, raw_scan);
+    bool in_mid = obstacle_in_range(raw_scan->angle_min + angle_side_mid, raw_scan->angle_max - angle_side_mid, dis_mid,
+                                    raw_scan);
+    bool in_far = obstacle_in_range(raw_scan->angle_min + angle_side_far, raw_scan->angle_max - angle_side_far, dis_far,
+                                    raw_scan);
+
+    //step 2 determine turn
+    //scan from side to see either left or right has more space, for certain distance
+    float distance_used;
+    float linear_speed = 13;
+    float angular_speed = 13;
+    //determine distance, velocity for different case
+    if (in_far) {
+        //certain linear;
+        vel_msg.linear.x = linear_speed;
+        vel_msg.angular.z = turn_left(raw_scan, dis_far) * angular_speed;
+    } else if (in_mid) {
+        //certain linear;
+        vel_msg.linear.x = linear_speed;
+        vel_msg.angular.z = turn_left(raw_scan, dis_mid)*angular_speed;
+    } else if (in_near) {
+        //certain linear;
+        vel_msg.linear.x = linear_speed;
+        vel_msg.angular.z = turn_left(raw_scan, dis_near)*angular_speed;
+    } else {
+        // no obstacle;
+        //certain linear;
+        vel_msg.linear.x = linear_speed;
+        vel_msg.angular.z = 0;
+    }
+    vel_msg.linear.y = 0;
+    vel_msg.linear.z = 0;
+    vel_msg.angular.x = 0;
+    vel_msg.angular.y = 0;
+
+    return vel_msg;
+}
+
 void LidarDecision::publishTwist(geometry_msgs::Twist twist){
```

Comment:
If you want to publish your twist message (you do), you'll need to pass it to this function.